builders/build/eth-api/dev-env/brownie.md

```diff
@@ -0,0 +1,348 @@
+        return value;
+    }
+}
+```
+
+## 编译Solidity {: #compiling-solidity }
+
+要编译合约您可以运行以下命令：
+
+```
+brownie compile
+```
+
+![Compile Brownie project](/images/builders/build/eth-api/dev-env/brownie/brownie-4.png)
+
+!!! 注意事项
+    首次编译合约时需要安装`solc`二进制文件，因此可能需要较长的时间。
+
+在成功编译后，您将会在`build/contracts`目录中找到您的文件。此文件包含智能合约的字节码和元数据，皆为`.json`文件。此外，`build`目录应当处于`.gitignore`文件之中，如果并非如此，请您将其移入。
+
+如果您希望设置指定的编译器版本，您可以在`brownie-config.yaml`文件中操作。请注意，如果您尚未创建此文件，您需要创建一个。接着，您可以指定编译器（如下所示）：
+
+```yaml
+compiler:
+  evm_version: london
+  solc:
+    version: 0.8.13
+    optimizer:
+      enabled: true
+      runs: 200
+```
+
+当Brownie接收到更改的消息时，您的合约才会重新编译。您可以运行以下命令强制执行新的编译：
+
+```
+brownie compile --all
+```
+
+## 部署合约 {: #deploying-the-contract }
+
+要部署`Box.sol`智能合约，您将需要撰写一个简单的部署脚本。您可以在`scripts`目录中创建一个新文件并将其命名为`deploy.py`：
+
+```
+cd scripts && touch deploy.py
+```
+
+接着，您将需要撰写您的部署脚本。您可以跟随以下步骤开始进行操作：
+
+1. 从`brownie`导入`Box`合约和`accounts`模块
+
+2. 使用`accounts.load()`加载您的账户，这将解密您的文件并根据提供的账户名称获取账户信息
+
+3. 使用实例中存在的`deploy`函数指定`from`账户和`gas_limit`以将其实例化
+
+```py
+# scripts/deploy.py
+from brownie import Box, accounts
+
+def main():
+    account = accounts.load('alice')
+    return Box.deploy(
+        {
+            'from': account,
+            'gas_limit': '200000'
+        }
+    )
+```
+
+您可以使用`run`命令并指定网络来部署`Box.sol`合约：
+
+=== "Moonbeam"
+    ```
+    brownie run scripts/deploy.py --network moonbeam-mainnet
+    ```
+
+=== "Moonriver"
+    ```
+    brownie run scripts/deploy.py --network moonriver-mainnet
+    ```
+
+=== "Moonbase Alpha"
+    ```
+    brownie run scripts/deploy.py --network moonbeam-testnet
+    ```
+
+=== "Moonbeam Dev Node"
+    ```
+    brownie run scripts/deploy.py --network moonbeam-dev
+    ```
+
+在数秒后，合约将会被部署且在终端中可见。
+
+![Deploy Brownie project](/images/builders/build/eth-api/dev-env/brownie/brownie-5.png)
+
+恭喜您，您的合约已上线！您可以储存地址，您将会在下个步骤中使用该地址并与之交互。
+
+## 与合约交互 {: #interacting-with-the-contract }
+
+要与您新部署的合约交互，您可以在Brownie `console`中运行以下命令：
+
+=== "Moonbeam"
+    ```
+    brownie console --network moonbeam-mainnet
+    ```
+
+=== "Moonriver"
+    ```
+    brownie console --network moonriver-mainnet
+    ```
+
+=== "Moonbase Alpha"
+    ```
+    brownie console --network moonbeam-testnet
+    ```
+
+=== "Moonbeam Dev Node"
+    ```
+    brownie console --network moonbeam-dev
+    ```
+
+随后，合约实例将会自动在终端中可见且可访问。它将会被打包在`ContractContainer`中，并允许您部署新的合约实例。要访问部署的合约，您可以使用`Box[0]`。您可以跟随以下步骤，调用`store`函数并将数值设置为`5`：
+
+1. 为合约创建一个变量
+   
+    ```
+    box = Box[0]
+    ```
+    
+2. 使用您的账户调用`store`函数并将数值设置为`5`
+   
+    ```
+    box.store(5, {'from': accounts.load('alice'), 'gas_limit': '50000'})
+    ```
+    
+3. 输入您账户的密码
+
+此交易将会由您的账户签署并传送至网络。现在，您可以通过以下步骤获取数据：
+
+1. 调用`retrieve`函数
+   
+    ```
+    box.retrieve({'from': accounts.load('alice')})
+    ```
+    
+2. 输入密码
+
+您将会看见`5`或是任何您先前储存的数据。
+
+![Interact with Brownie project](/images/builders/build/eth-api/dev-env/brownie/brownie-6.png)
+
+恭喜您！您已经成功通过Brownie部署合约并与之交互！
```

Comment:
the `### Using a Script` section is missing: https://github.com/PureStake/moonbeam-docs/pull/283/files#